tests/test_edgeql_data_migration.py

```diff
@@ -6903,10 +6897,11 @@ async def test_edgeql_migration_eq_collections_06(self):
         )
 
     @test.xfail('''
-        The "complete" flag is not set even though the DDL from
-        "proposed" list is used.
+        ISE: column <blah> cannot be cast automatically to type
```

Comment:
This type of tests needs to be converted to use the `DESCRIBE CURRENT MIGRATION` mechanism, since altering a type requires user input.

Reply:
Clarification: no need to do the conversion in this PR.

Reply:
What does "requires" mean exactly here? We do all sorts of wild stuff in this file without describe current migration, and using POPULATE MIGRATION; COMMIT MIGRATION; works for most cases too?

This version ought to produce a better error than an ISE, too, right?

Reply:
> This version ought to produce a better error than an ISE, too, right?

Yes. `POPULATE MIGRATION` should abort if user input is required to properly generate a migration.  What you're seeing currently is a Postgres equivalent of such an error.  I'm working on this bit  right now.

Reply:
Even `migrate` here kind of simulates user input, though, right? It repeatedly selects the proposed DDL.

Reply:
> kind of simulates user input, though, right?

It simulates agreement with proposed DDL, but here there is no way to even generate any propositions, since we don't know how to convert from one data to another if there's no defined cast.  Similarly, when a `required` property is added, we must know the value to use to back-populate existing entries.